tests/utils/test_file_utils.py

```diff
@@ -364,3 +365,35 @@ def test_shutil_copytree_without_file_permissions(tmp_path):
     assert set(os.listdir(dst_dir.joinpath("subdir"))) == {"subdir-file.txt"}
     assert dst_dir.joinpath("subdir/subdir-file.txt").read_text() == "testing 123"
     assert dst_dir.joinpath("top-level-file.txt").read_text() == "hi"
+
+
+def test_get_total_size_basic(tmp_path):
+    root = str(tmp_path)
+    subdir = os.path.join(root, "subdir")
+    os.mkdir(subdir)
+
+    def generate_file(path, size_in_bytes):
+        with open(path, "wb") as fp:
+            fp.write(b"\0" * size_in_bytes)
+
+    file_size_map = {"file1.txt": 11, "file2.txt": 23}
+    for name, size in file_size_map.items():
+        path = os.path.join(root, name)
+        generate_file(path, size)
+    generate_file(os.path.join(subdir, "file3.txt"), 22)
+    assert get_total_file_size(root) == 56
+    assert get_total_file_size(subdir) == 22
+
+    path_not_exists = os.path.join(root, "does_not_exist")
+    with pytest.raises(
+        MlflowException,
+        match="does not exist.",
+    ):
+        get_total_file_size(path_not_exists)
+
+    path_file = os.path.join(root, "file1.txt")
+    with pytest.raises(
+        MlflowException,
+        match="is not a directory.",
+    ):
+        get_total_file_size(path_file)
```

Comment:
/ai rewrite this using pathlib.Path methods. `tmp_path` is a pathlib.Path object.